code/controllers/subsystem/processing/greyscale.dm

```diff
@@ -27,12 +27,68 @@ PROCESSING_SUBSYSTEM_DEF(greyscale)
 		var/datum/greyscale_config/config = configurations[greyscale_type]
 		config.CrossVerify()
 
+	if(CONFIG_GET(flag/generate_assets_in_init))
+		ExportMapPreviews()
+
 	return SS_INIT_SUCCESS
 
 /datum/controller/subsystem/processing/greyscale/proc/RefreshConfigsFromFile()
 	for(var/i in configurations)
 		configurations[i].Refresh(TRUE)
 
+/datum/controller/subsystem/processing/greyscale/proc/ExportMapPreviews()
+	// Put subtypes before their parent or the parent file will take all the generated icons
+	var/static/list/types_that_get_their_own_file = list(
+		"turfs" = /turf, // None of these yet but it's harmless to be prepared
+		"mobs" = /mob, // Ditto
+		"clothing" = /obj/item/clothing,
+		"items" = /obj/item,
+		"objects" = /obj,
+	)
+
+	var/list/handled_types = list()
+	for(var/filename in types_that_get_their_own_file)
+		var/type_to_export = types_that_get_their_own_file[filename]
+		handled_types += ExportMapPreviewsForType(filename, type_to_export, handled_types)
+
+	ExportMapPreviewsForType("unsorted", /atom, handled_types)
+
+/datum/controller/subsystem/processing/greyscale/proc/ExportMapPreviewsForType(filename, atom/atom_typepath, list/type_blacklist)
+	var/list/handled_types = list()
+	var/list/icons = list()
+	for(var/atom/fake as anything in subtypesof(atom_typepath))
+		CHECK_TICK
+		if(type_blacklist && type_blacklist[fake])
+			continue
+		handled_types[fake] = TRUE
+		var/greyscale_config = fake::greyscale_config
+		var/greyscale_colors = fake::greyscale_colors
+		if(!greyscale_config || !greyscale_colors)
+			continue
+		var/icon/map_icon = GetColoredIconByType(greyscale_config, greyscale_colors)
+		if(!(fake::post_init_icon_state in map_icon.IconStates()))
+			stack_trace("GAGS configuration missing icon state needed to generate mapping tool graphic for '[fake]'. Make sure the right greyscale_config is set up.")
+			continue
+		map_icon = icon(map_icon, fake::post_init_icon_state)
+		icons["[fake]"] = map_icon
+
+	var/icon/holder = icon('icons/testing/greyscale_error.dmi')
+	for(var/state in icons)
+		CHECK_TICK
+		holder.Insert(icons[state], state)
```

Comment:
this strikes me as highly likely to give bad icon errors due to both the check tick and the high volume of icons being processed here. can't recall if Insert requires nonstatic icons tho so maybe the problem is limited just to the CHECK_TICK

Reply:
I did break up the icons a bit into different files partially to avoid bad icon errors, but you're right that check tick might allow other icon operations to run and cause a bad icon error here. Want me to just remove them all since this shouldn't ever run on live anyway?

Reply:
I think it'd be best to just remove checktick since it'll only ever waste time as you say, and it risks a static holder